dm/simulator/internal/mcp/uk.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mcp
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// UniqueKey is the data structure describing a unique key.
+type UniqueKey struct {
+	// It inherits a RWMutex, which is used to modify the metadata inside the UK struct.
+	sync.RWMutex
+	// rowOPLock is the lock for operating on the corresponding row of the unique key
+	rowOPLock sync.Mutex
```

Comment:
What's RowOperation 🤔

Reply:
A row operation is a DML operation on a specific row: UPDATE or DELETE.  In simulation process, If one transaction is updating a row in MCP, I don't want other workers to also modify/delete this row.  So I introduce a lock here.

Reply:
After further investigation, I think the row operation lock is not needed for the sake of performance.  I'll remove that.